src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -812,6 +812,8 @@ private Task WriteSuffix()
             var responseHeaders = FrameResponseHeaders;
             var hasConnection = responseHeaders.HasConnection;
             var connectionOptions = hasConnection ? FrameHeaders.ParseConnection(responseHeaders.HeaderConnection) : ConnectionOptions.None;
+            var hasTransferEncoding = responseHeaders.HasTransferEncoding;
+            var transferCoding = hasTransferEncoding ? FrameHeaders.GetFinalTransferCoding(responseHeaders.HeaderTransferEncoding) : TransferCoding.None;
```

Comment:
Nit: HasConnection and HasContentLength both simply call StringValues.Count. It would be simpler to have ParseConnection and GetFinalTransferCoding check the Count themselves and return None rather than doing the Has checks here.